bundles/framework/timeseries/WMSAnimator.js

```diff
@@ -0,0 +1,180 @@
+            return times;
+        },
+        init: function () { },
+        /**
+         * @method getCurrentTime
+         * Returns current selected time instant
+         * @return {String} current time as ISO-string
+         */
+        getCurrentTime: function () {
+            return this._currentTime;
+        },
+        /**
+         * @method getSubsetRange
+         * Returns current selected time range subset
+         * @return {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        getSubsetRange: function () {
+            return this._subsetRange.slice();
+        },
+        /**
+         * @method getSubsetRange
+         * Set current selected time range subset
+         * @param {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        setSubsetRange: function (range) {
+            this._subsetRange = range;
+        },
+        /**
+         * @method requestNewTime
+         * Requests change in current selected time
+         * @param {String} newTime change current time to this value(ISO string)
+         * @param {String} nextTime time value at next animation frame(ISO string). Can be null if not animating
+         * @param {function} doneCallback callback that will be called after new time has been loaded
+         */
+        requestNewTime: function (newTime, nextTime, doneCallback) {
+            var me = this;
+            this._currentTime = newTime;
+            var requestBuilder = this._sandbox.getRequestBuilder('MapModulePlugin.MapLayerUpdateRequest');
+            if (requestBuilder) {
+                this._isLoading = true;
+                this._doneCallback = doneCallback;
+                if (nextTime) {
+                    this._bufferImages(this._mapModule.getLayerTileUrls(this._layer.getId()), nextTime, function (success) {
+                        me._isBuffering = false;
+                        me._resolveWait();
+                    });
+                }
+                var request = requestBuilder(this._layer.getId(), true, { "TIME": newTime });
+                this._sandbox.request(this, request);
+            }
+        },
+        /**
+         * @method _bufferImages
+         * @private
+         * Preload tile images that will be used in next animation frame
+         * @param {String[]} urls urls of tile images
+         * @param {String} nextTime the time instant to request tiles for
+         * @param {Function} callback is called when loading is ready or 5000 ms timeout reached
+         */
+        _bufferImages: function (urls, nextTime, callback) {
+            this._isBuffering = true;
+            var imgCount = urls.length;
+            if (imgCount === 0) {
+                callback(true);
+                return;
+            }
+            var aborted = false;
+            var timeout = setTimeout(function () {
+                aborted = true;
+                callback(false);
+            }, 5000);
```

Comment:
Should the timeout be configurable with layer.attributes.imageTimeout or similar and default to 5 seconds if not configured?

Reply:
Maybe not since the timeout is just for buffering. The caller doesn't seem to mind if buffering timed out.

Reply:
I would add the configuration in this case when/if it's needed in the future. _bufferImages() is a private method and in the way it's currently used, it doesn't really make a difference if the buffering failed or succeeded. There just needs to be some reasonable timeout so that the animation can progress in any case.

Reply:
Ok, lets keep it as is